### PR TITLE
feat: Set the DISABLE_DEX environment variable to true in CSV

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind enhancement

**What does this PR do / why we need it**:
This PR set the DISABLE_DEX environment variable to true in CSV.  By doing so, the Dex workload is enabled and the operator controller reconciler dex/oidc configuration.

**Test acceptance criteria**:
* [x] E2E Test

**How to test changes / Special notes to the reviewer**:
run the operator locally using `operator-sdk run local`
After a couple of minutes, verify the install installation is successful and pod is up and running.

### CODE REVIEW

**Comment by iam-veeramalla** [comment]
@amitkrout is this an expected failure ?

**Comment by iam-veeramalla** [comment]
@jaideepr97  @wtam2018 can you please review and approve the PR

**Comment by jaideepr97** [comment]
Hey @iam-veeramalla, thanks for the PR 

based on the conversation with shoubhik, don't we need a check to see if it's a fresh installation before enabling dex?

**Comment by iam-veeramalla** [comment]
@jaideepr97 I thought we can take that as a different story. It is better to handle it as a different story as we need to think about the upgrade scenarios for both default and non-default instances and write some tests.

File: test/e2e/gitopsservice_test.go [hunk 1]
```diff
@@ -62,6 +62,12 @@ var _ = Describe("GitOpsServiceController", func() {
 
 		It("Argo CD instance is created", func() {
 			checkIfPresent(types.NamespacedName{Name: argoCDInstanceName, Namespace: argoCDNamespace}, argoCDInstance)
+			checkIfPresent(types.NamespacedName{Name: defaultApplicationControllerName, Namespace: argoCDNamespace}, &appsv1.StatefulSetList{})
+			checkIfPresent(types.NamespacedName{Name: defaultApplicationSetControllerName, Namespace: argoCDNamespace}, &appsv1.Deployment{})
+			checkIfPresent(types.NamespacedName{Name: defaultDexInstanceName, Namespace: argoCDNamespace}, &appsv1.Deployment{})
+			checkIfPresent(types.NamespacedName{Name: defaultRedisName, Namespace: argoCDNamespace}, &appsv1.Deployment{})
+			checkIfPresent(types.NamespacedName{Name: defaultRepoServerName, Namespace: argoCDNamespace}, &appsv1.Deployment{})
+			checkIfPresent(types.NamespacedName{Name: defaultRepoServerName, Namespace: argoCDNamespace}, &appsv1.Deployment{})
```
Comment (jaideepr97) [inline test/e2e/gitopsservice_test.go:70]: I think one of these is supposed to be gitops server?

Comment (iam-veeramalla) [reply test/e2e/gitopsservice_test.go]: Nice catch , Thanks :)

**Pull request merged**